src/script/standard.cpp

```diff
@@ -386,6 +396,17 @@ bool IsValidDestination(const CTxDestination& dest) {
     return ret;
 }
 
+void TaprootSpendData::Merge(TaprootSpendData other)
+{
+    if (internal_key.IsNull() && !other.internal_key.IsNull()) {
```

Comment:
Asserting sounds dangerous, as I'd expect TaprootSpendData to at some point be untrusted data (e.g. gathered from PSBT). I agree it's a bit ugly to just prefer one over the other in case of a contradiction, and I also don't have a better solution. Would you be ok with just a TODO here?